From c622cbf874477bd2427711e85d5ca9a4aea4cfe3 Mon Sep 17 00:00:00 2001
From: P Praneesh <quic_ppranees@quicinc.com>
Date: Mon, 10 Jun 2024 20:11:30 +0530
Subject: [PATCH] wifi: ath12k: remove CAC running check in rx path

During CAC, transmission is restricted but current implementation checks
whether CAC is running for each rx packet processing which is
unnecessary. Remove the CAC running check in the per-packet processing
path during CAC. Allow the driver to continue received packet processing
even when tx is not allowed.

Patch Dependency: None

Signed-off-by: P Praneesh <quic_ppranees@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/dp_rx.c | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/drivers/net/wireless/ath/ath12k/dp_rx.c b/drivers/net/wireless/ath/ath12k/dp_rx.c
index b7c8bf0..7266d1d 100644
--- a/drivers/net/wireless/ath/ath12k/dp_rx.c
+++ b/drivers/net/wireless/ath/ath12k/dp_rx.c
@@ -3429,11 +3429,6 @@ static void ath12k_dp_rx_process_received_packets(struct ath12k_base *ab,
 			continue;
 		}
 
-		if (unlikely(test_bit(ATH12K_CAC_RUNNING, &ar->dev_flags))) {
-			dev_kfree_skb_any(msdu);
-			continue;
-		}
-
 		ret = ath12k_dp_rx_process_msdu(ar, msdu, msdu_list, &rx_info,
 						&fast_rx);
 		if (ret) {
-- 
2.34.1

